src/integration-tests/src/observability/auth.rs

```diff
@@ -0,0 +1,335 @@
+    /// credentials refreshed.
+    pub fn new(credentials: Credentials) -> Self {
+        let (tx, rx) = watch::channel(None);
+        tokio::spawn(refresh_task(credentials, tx));
+        Self { rx }
+    }
+}
+
+impl Interceptor for CloudTelemetryAuthInterceptor {
+    fn call(&mut self, mut request: Request<()>) -> Result<Request<()>, Status> {
+        // Read the latest headers from the watch channel.
+        let rx_ref = self.rx.borrow();
+        let metadata = rx_ref.as_ref().ok_or_else(|| {
+            // If the first refresh hasn't completed yet, fail the request.
+            // The OTLP exporter is expected to handle this transient failure
+            // with its built-in retry mechanism.
+            Status::unauthenticated("GCP credentials not yet available")
+        })?;
+
+        for entry in metadata.iter() {
+            match entry {
+                tonic::metadata::KeyAndValueRef::Ascii(key, value) => {
+                    request.metadata_mut().insert(key.clone(), value.clone());
+                }
+                tonic::metadata::KeyAndValueRef::Binary(key, value) => {
+                    request
+                        .metadata_mut()
+                        .insert_bin(key.clone(), value.clone());
+                }
+            }
+        }
+        Ok(request)
+    }
+}
+
+/// Background task that periodically refreshes credentials and sends them
+/// to the interceptor via a watch channel.
+async fn refresh_task(credentials: Credentials, tx: watch::Sender<Option<MetadataMap>>) {
+    let mut last_etag: Option<EntityTag> = None;
+    loop {
+        let mut extensions = http::Extensions::new();
+        if let Some(etag) = last_etag.clone() {
+            extensions.insert(etag);
+        }
+
+        match credentials.headers(extensions).await {
+            Ok(CacheableResource::New { entity_tag, data }) => {
+                let mut metadata = MetadataMap::new();
+                for (name, value) in data.iter() {
+                    if let (Ok(key), Ok(val)) = (
+                        tonic::metadata::MetadataKey::from_bytes(name.as_str().as_bytes()),
+                        tonic::metadata::MetadataValue::try_from(value.as_bytes()),
+                    ) {
+                        metadata.insert(key, val);
+                    } else {
+                        // Skip invalid headers. This can happen if the header name or value
+                        // contains characters that are not allowed in HTTP/2 metadata
+                        // (e.g., non-ASCII characters in values without -bin suffix).
+                    }
+                }
+
+                if tx.send(Some(metadata)).is_err() {
+                    // Receiver dropped (all interceptors are gone), stop task.
+                    break;
+                }
+                last_etag = Some(entity_tag);
+                sleep(REFRESH_INTERVAL).await;
```

Comment:
The refresh interval seems too long. If the credentials have expired the requests will fail for 5 minutes?